### PR TITLE
Authentication with firebase

### PR DESCRIPTION
**Pull request by fava2020**
Screenshot 2024-04-09 at 8 15 53 p m
Screenshot 2024-04-09 at 8 15 44 p m

### CODE REVIEW

File: app-quizz/.env
Comment (miguelromerogutierrez) [inline app-quizz/.env:1]: This file must be hided by .gitignore

File: app-quizz/src/app/authentication/sign-in/page.jsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        } else if (values.password.length <= 8) {
+            errors.password = "Password length must be more than 7"
+        }
+
+        return errors;
+    }
+
+    return (
+    <div className="min-h-screen flex items-center justify-center bg-white">
+      <div className="bg-teal-100 p-10 rounded-lg shadow-xl w-96">
```
Comment (miguelromerogutierrez) [inline app-quizz/src/app/authentication/sign-in/page.jsx:51]: Can we set the tab-spacing for 2 spaces, please?

Comment (miguelromerogutierrez) [reply app-quizz/src/app/authentication/sign-in/page.jsx:51]: Look at this https://gist.github.com/karpolan/2c573b5767bc9b65db9936c7fad4daac

File: app-quizz/src/app/authentication/sign-in/page.jsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                                </p>
+                            </div>
+                        </div>
+                        <div className="flex flex-row text-l pl-10 font-sans text-sm text-teal-900 justify-end">
+                            <div>No account yet?&nbsp;</div>
+                            <div><Link href="/authentication/sign-up" className="text-teal-900 text-secondary">Sign up</Link></div>
+                        </div>
+
+                        <div className="flex flex-row text-l pb-8 pl-10 justify-end">
+                            <Button handlerEvent={() => {}} title={loading ? "Logging in ..." : " Login "}></Button>
```
Comment (miguelromerogutierrez) [inline app-quizz/src/app/authentication/sign-in/page.jsx]: Why an empty `handlerEvent`?

File: app-quizz/src/app/authentication/sign-in/page.jsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                    errors,
+                    touched,
+                    handleChange,
+                    handleBlur,
+                    handleSubmit,
+                    isSubmitting
+                }) => (
+                    <Form className="mt-8 space-y-6" >
+                        <div className=" space-y-6 rounded-md shadow-sm">
+                            <div>
```
Comment (miguelromerogutierrez) [inline app-quizz/src/app/authentication/sign-in/page.jsx]: Maybe is a good idea to isolate this building block into a UIField component:
```
export function UIField({label, placeholder, id, name, ....}) {
  return               <div>
                                <label htmlFor="email-address" className="sr-only">
                                    Email address
                                </label>
                                <Field
                                    type="email"
                                    id="email"
                                    name="email"
                                    value={values.email}
                                    onChange={handleChange}
                                    className="w-full p-3 bg-white rounded outline-none text-teal-900 placeholder-teal-700"
                                    placeholder="Email address"
                                />

                                <p className='text-xs text-red-600'>
                                    {errors.email && touched.email && errors.email}
                                </p>
                            </div>
}
```

then you can use it simply:

```
<>
  <UIField name="email" label="Email" .../>
  <UIField name="password" label="Password" ... />
</>
```

File: app-quizz/src/app/authentication/sign-in/page.jsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            sessionStorage.setItem('user', true);
+            router.push('/')
+        }catch(e){
+            setLoading(false);
+            setErrors(e.message);
+            console.error(e);
+        }
+    };
+
+    const validateForm = (values) => {
```
Comment (miguelromerogutierrez) [inline app-quizz/src/app/authentication/sign-in/page.jsx]: move out this function. Maybe in a utils.ts file along this component

File: app-quizz/src/app/authentication/sign-up/page.jsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        setEmail('');
+        setPassword('')
+
+    } catch(e){
+        console.error(e)
+    }
+  };
+
+  return (
+    <div className="min-h-screen flex items-center justify-center bg-white">
```
Comment (miguelromerogutierrez) [inline app-quizz/src/app/authentication/sign-up/page.jsx:34]: here is missing consistency, use Formik and the fields, also validate the fields same as sign-in page

**Review by miguelromerogutierrez — CHANGES_REQUESTED** [review]

File: app-quizz/src/app/authentication/sign-in/page.jsx (hunk 2 above)
Comment (fava2020) [reply app-quizz/src/app/authentication/sign-in/page.jsx]: My mistake 😂

File: app-quizz/src/app/authentication/sign-in/page.jsx (hunk 1 above)
Comment (fava2020) [reply app-quizz/src/app/authentication/sign-in/page.jsx:51]: I will take a look

File: app-quizz/src/app/authentication/sign-in/page.jsx (hunk 3 above)
Comment (fava2020) [reply app-quizz/src/app/authentication/sign-in/page.jsx]: Nice

File: app-quizz/src/app/authentication/sign-up/page.jsx (hunk 5 above)
Comment (fava2020) [reply app-quizz/src/app/authentication/sign-up/page.jsx:34]: I will add here it

File: app-quizz/.env
Comment (fava2020) [reply app-quizz/.env:1]: @miguelromerogutierrez I am unsure why these files are not being ignored.

**Review by miguelromerogutierrez — APPROVED** [review]

**Comment by miguelromerogutierrez** [comment]
Looks really good

**Pull request merged**